### ISSUE TITLE
Adding a sub-task via drag-drop

### ISSUE THREAD

**Issue by plamkas** [johannesjo/super-productivity#905]
Sometimes I create a couple of tasks only to find out that somehow they all have the same origin. I could not find a way to create a new task and to assign the previously created tasks a sub-task to that new one.

**Comment by JLuisGT** [comment]
+1  :D

**Comment by johannesjo** [comment]
Related to #235

**Comment by JoseOlin** [comment]
I think this is still relevant ;)

**Comment by ced75** [comment]
I agree. It is possible to define a subtask as a task, but the opposite is not possible.
It would be nice if we could move a task as a subtask of another existing task.

**Comment by ced75** [comment]
Still relevant :)

**Comment by Pictor13** [comment]
> It would be nice if we could move a task as a subtask of another existing task.

That's not possible as a task might have a sub-task already, so converting the task to sub-task would provoke a hierarchy tree with 2 levels depth, and unfortunately sub-tasks of sub-tasks are not supported by Super Productivity (#643).
So I think that's the blocker: _what should be done with a sub-task when its parent is itself being converted to a sub-task?_

That's a pity as multiple level subtask or proper referencing of tasks is a fundamental feature in my workflow.
Other than that, the app would be perfect.

At least referencing other tasks would make it workable for more complex/granular projects.

**Comment by jschbrt** [comment]
Any update on this? - Is maybe someone working on it currently?

**Comment by nicola-lunghi** [comment]
+1 here

would be useful to drag tasks into another task to convert it to subtask or drag it out to make it a parent task if it gets too big

**Comment by ItsIgnacioPortal** [comment]
> _what should be done with a sub-task when its parent is itself being converted to a sub-task?_

To me, the answer is clear. Transform the orphan subtask into a regular, root-level task.

**Comment by ItsIgnacioPortal** [comment]
Still relevant

**Comment by ItsIgnacioPortal** [comment]
Still relevantes.

**Comment by dahaupt** [comment]
Still looking forward to this.

**Comment by Nelvin** [comment]
+1 to this feature - I just came back to give superproductivity another try and started by creating a bunch of tasks I wanted to group/organize after I've added them which, to my surprise, was not possible.

The most simple way to handle tasks containing subtasks already would be to just not allow those to be transformed into subtasks. Transforming existing subtasks into parent tasks is another option (as menationed by @ItsIgnacioPortal already) but as we can do this manually already, that's much less of an issue - the difference is, requiring a few seconds vs impossible.

**Comment by gergof** [comment]
+1 would like to see this implemented.

**Comment by sharktide** [comment]
consider adding something on https://gitpay.me/#/task/1146/adding-a-sub-task-via-drag-drop (thats how I found out about this anyway). More people might want to contribute this fix (possibly me as well)

**Comment by bytrangle** [comment]
So many requests, but the requirements are not clear. How exactly does one create a subtask via drag and drop? Let's say I have two tasks: "Task 1" and "Task 2". What is needed to turn "Task 2" into a subtask for "Task 1"? In Todoist, it will require dragging "Task 2" right under "Task 1", and **to the right**. Without the second action, it would just be simple reordering.

**Comment by blefuscu** [comment]
Still a think 👍

**Comment by Squirrels-088** [comment]
Still relevant. I was shocked it couldn't do this!

**Comment by bytrangle** [comment]
@Squirrels-088: perhaps consider contributing to bounty hunter to that someone qualified will take on this task?

**Comment by NoNeed2026** [comment]
Definitely relevant, would be great if Johannes could surprise us with this feature ;)

**Comment by impronta48** [comment]
I was looking for this!

**Comment by erlouche** [comment]
Also looking for this feature.
